builtin/plugins/dposv3/dpos_test.go

```diff
@@ -1299,6 +1299,175 @@ func TestClaimRewardsFromMultipleValidators(t *testing.T) {
+	}
+
+	candidates, err := dpos.ListCandidates(pctx)
+	require.Nil(t, err)
+	assert.Equal(t, 3, len(candidates))
+
+	err = dpos.UnregisterCandidate(pctx.WithSender(addr1))
+	require.Nil(t, err)
+
+	candidates, err = dpos.ListCandidates(pctx)
+	require.Nil(t, err)
+	assert.Equal(t, 3, len(candidates))
+
+	validators, err = dpos.ListValidators(pctx)
+	require.Nil(t, err)
+	assert.Equal(t, 3, len(validators))
+
+	require.NoError(t, elect(pctx, dpos.Address))
+	pctx.SetTime(pctx.Now().Add(time.Duration(cycleLengthSeconds) * time.Second))
+
+	rewardFromAddr1Before, err := dpos.CheckRewardDelegation(pctx.WithSender(delegatorAddress1), &addr1)
+	require.NoError(t, err)
+	require.True(t, rewardFromAddr1Before.Amount.Value.Cmp(common.BigZero()) > 0)
+
+	rewardFromAddr2, err := dpos.CheckRewardDelegation(pctx.WithSender(delegatorAddress1), &addr2)
+	require.True(t, rewardFromAddr2.Amount.Value.Cmp(&rewardFromAddr1Before.Amount.Value) == 0)
+	require.NoError(t, err)
+
+	rewardFromAddr3, err := dpos.CheckRewardDelegation(pctx.WithSender(delegatorAddress1), &addr3)
+	require.NoError(t, err)
+	require.True(t, rewardFromAddr2.Amount.Value.Cmp(&rewardFromAddr3.Amount.Value) == 0)
+
+	require.NoError(t, elect(pctx, dpos.Address))
+	pctx.SetTime(pctx.Now().Add(time.Duration(cycleLengthSeconds) * time.Second))
+
+	candidates, err = dpos.ListCandidates(pctx)
+	require.Nil(t, err)
+	assert.Equal(t, 2, len(candidates))
+
+	validators, err = dpos.ListValidators(pctx)
+	require.Nil(t, err)
+	assert.Equal(t, 2, len(validators))
+
+	// Check that delegation reward after addr1 Unregistered from candidate list remain the same.
+	rewardFromAddr1, err := dpos.CheckRewardDelegation(pctx.WithSender(delegatorAddress1), &addr1)
+	require.NoError(t, err)
+	require.True(t, rewardFromAddr1.Amount.Value.Cmp(&rewardFromAddr1Before.Amount.Value) == 0)
+
+	// And delegation reward from addr2 & addr3 still increasing.
+	rewardFromAddr2, err = dpos.CheckRewardDelegation(pctx.WithSender(delegatorAddress1), &addr2)
+	require.NoError(t, err)
+	rewardFromAddr3, err = dpos.CheckRewardDelegation(pctx.WithSender(delegatorAddress1), &addr3)
+	require.NoError(t, err)
+	require.True(t, rewardFromAddr2.Amount.Value.Cmp(&rewardFromAddr3.Amount.Value) == 0)
+	require.True(t, rewardFromAddr2.Amount.Value.Cmp(&rewardFromAddr1.Amount.Value) > 0)
+
+	pctx.SetFeature(features.DPOSVersion3_6, true)
+	require.True(t, pctx.FeatureEnabled(features.DPOSVersion3_6, false))
+	// User claims the rewards they expected with 1 call.
+	// They are also able to get the amount that was claimed in the same call
+	claimedAmt, err := dpos.ClaimDelegatorRewards(pctx.WithSender(delegatorAddress1))
+
+	balanceBeforeUnbond, err := coinContract.BalanceOf(contractpb.WrapPluginContext(coinCtx), &coin.BalanceOfRequest{
+		Owner: delegatorAddress1.MarshalPB(),
+	})
+	require.Nil(t, err)
+
+	// execute elections to make funds available in the acc balance
```

Comment:
Before this election load all the reward delegations for this delegator and add them all up, then compare the sum to the `balanceAfterUnbond - balanceBeforeUnbond`.